### ISSUE TITLE
Is there any way to returning value from c# to js

### ISSUE THREAD

**Issue by berk0ayvaz** [nmoschkin/MAUIWebViewExample#7]
Hello im traying to returning value from c#.

Like 

<script>
var a = invokeCSharpAction(counter++);

alert(a);

<script>

**Comment by nmoschkin** [comment]
No.  But you can create a callback.

You can write a method that receive the return in Javascript, and use the Javascript invoke feature from C# to send the response back to the Javascript.

You could even use a tokening mechanism to make sure the response can be matched with the request.

In Javascript:

InvokeCSharpAction(JSON.stringify({token: token, data:data}));

const csresult = (token, r) => {
   
};

in C#

OnCSharpEvent(data){
  // do stuff
  var obj = JsonConvert.DeserializeObject<MyDataClass>(data);
  var token = obj.uuid;
  var results = { something you do with the data };
  webBrowser.EvaluateJavascriptAsync(token, results);
}

**Comment by tedchirvasiu** [comment]
Based on @nmoschkin's suggestion, I implemented an example:
https://github.com/tedchirvasiu/MAUIWebViewExample
https://github.com/tedchirvasiu/MAUIWebViewExample/commit/31bfb68f37038d7dfc1401e326ec11cc7404bbcc

I tried not to mess with the internal code and only operate on the MainPage example. The only change to the internal code I did was to remove the JSON.stringify from HybridWebViewHandler on Windows as that would encode it twice.

In `MainPage` I made `MyWebView_JavaScriptAction` act like a reducer in Redux, where the Payload it receives is a serialized JSON object of the form:
```csharp
public class JavascriptAction {
    public string Type { get; set; }
    public JsonNode? Payload { get; set; }
    public string? ResponseToken { get; set; }
}
```
So now `MyWebView_JavaScriptAction` routes the action based on the action type.
https://github.com/tedchirvasiu/MAUIWebViewExample/blob/31bfb68f37038d7dfc1401e326ec11cc7404bbcc/WebViewHostExample/MainPage.xaml.cs#L48

Then, based on whether a ResponseToken is present or not, it invokes a receiver function on the JavaScript side with the response.

On the JavaScript side you'll see a separate script block with a mini clojure called "csharp". This encapsulates 4 methods:
1. invoke(action, data) => void - this is just a raw wrapper around invokeCSharpAction, it does not track a response.
2. call(action, data) => Promise\<string\> - this returns a promise which gets resolved with the response from the native code or rejected if the native code threw an error
3. receiveResponse(token, data) - this is called by the native code to return the response on success
4. receiveError(token) - this is called by the native code to indicate an error occured

So, in short, in modern JavaScript you could simply call:
```Javascript
const result = await csharp.call('my_action_name', data);
```

Warning, only briefly tested on Windows and Android.

@nmoschkin I could submit a PR unless you feel it ruins the simplicity of the current example.

**Comment by nmoschkin** [comment]
This example is extremely basic.  It's bare and skeletal, and from here we can go in many fancy directions. 

We could even write code to bind types, etc... lots of different and creative things.  

So the question is, do your changes correct existing functionality, or introduce new functionality?  Because this is not trying to be a library.  Just a very barebones, core, basic example.  I expect that people fork it, and do what they like with it.
